staging/src/k8s.io/apiserver/pkg/server/filters/timeout_test.go

```diff
@@ -255,42 +253,12 @@ func TestTimeoutHeaders(t *testing.T) {
 	res.Body.Close()
 }
 
-func captureStdErr() (func() string, func(), error) {
+func TestErrConnKilled(t *testing.T) {
 	var buf bytes.Buffer
-	reader, writer, err := os.Pipe()
-	if err != nil {
-		return nil, nil, err
-	}
-	stderr := os.Stderr
-	readerClosedCh := make(chan struct{})
-	stopReadingStdErr := func() string {
-		writer.Close()
-		<-readerClosedCh
-		return buf.String()
-	}
-	klog.LogToStderr(true)
-	cleanUp := func() {
-		os.Stderr = stderr
-		klog.LogToStderr(false)
-		stopReadingStdErr()
-	}
-	os.Stderr = writer
-	go func() {
-		io.Copy(&buf, reader)
-		readerClosedCh <- struct{}{}
-		close(readerClosedCh)
-	}()
-	klog.LogToStderr(true)
+	klog.SetOutput(&buf)
+	klog.LogToStderr(false)
+	defer klog.LogToStderr(true)
```

Comment:
Just to double check - are we running all tests generally with also logging to stderr?
If so - this looks fine, but I want to double check we won't accidentally break some other unrelated tests.

@pohly @aojea ?

Reply:
should not be broken before this PR too?

Reply:
ok - fair